// @calcom/atoms/dist/Card-AX0eZjOC.js
import { aq as N, az as m, h as u, aA as p } from "./index-6PQ0n_zn.js";
import { jsxs as d, jsx as r } from "react/jsx-runtime";
const i = p("", {
  // Variants won't have any style by default. Style will only be applied if the variants are combined.
  // So, style is defined in compoundVariants.
  variants: {
    variant: {
      basic: "",
      ProfileCard: "",
      SidebarCard: ""
    },
    structure: {
      image: "",
      card: "",
      title: "",
      description: ""
    }
  },
  compoundVariants: [
    // Style for Basic Variants types
    {
      variant: "basic",
      structure: "image",
      className: "w-10 h-auto"
    },
    {
      variant: "basic",
      structure: "card",
      className: "p-5"
    },
    {
      variant: "basic",
      structure: "title",
      className: "text-base mt-4"
    },
    {
      variant: "basic",
      structure: "description",
      className: "text-sm leading-[18px] text-subtle font-normal"
    },
    // Style for ProfileCard Variant Types
    {
      variant: "ProfileCard",
      structure: "image",
      className: "w-9 h-auto rounded-full mb-4s"
    },
    {
      variant: "ProfileCard",
      structure: "card",
      className: "w-80 p-4 hover:bg-subtle"
    },
    {
      variant: "ProfileCard",
      structure: "title",
      className: "text-base"
    },
    {
      variant: "ProfileCard",
      structure: "description",
      className: "text-sm leading-[18px] text-subtle font-normal"
    },
    // Style for SidebarCard Variant Types
    {
      variant: "SidebarCard",
      structure: "image",
      className: "w-9 h-auto rounded-full mb-4s"
    },
    {
      variant: "SidebarCard",
      structure: "card",
      className: "w-full p-3 border border-subtle"
    },
    {
      variant: "SidebarCard",
      structure: "title",
      className: "text-sm font-cal"
    },
    {
      variant: "SidebarCard",
      structure: "description",
      className: "text-xs text-default line-clamp-2"
    }
  ]
});
function g({
  image: f,
  title: h,
  icon: b,
  description: c,
  variant: s,
  actionButton: e,
  containerProps: t,
  imageProps: a,
  mediaLink: x,
  thumbnailUrl: v,
  learnMore: l
}) {
  const C = l && l.href.startsWith("https") ? "a" : N;
  return /* @__PURE__ */ d("div", {
    className: m(t == null ? void 0 : t.className, i({
      variant: s,
      structure: "card"
    }), "bg-default border-subtle text-default flex flex-col justify-between rounded-md border"),
    "data-testid": "card-container",
    ...t,
    children: [/* @__PURE__ */ d("div", {
      children: [b && b, f && /* @__PURE__ */ r("img", {
        src: f,
        alt: a == null ? void 0 : a.alt,
        className: m(a == null ? void 0 : a.className, i({
          variant: s,
          structure: "image"
        })),
        ...a
      }), /* @__PURE__ */ r("h5", {
        title: h,
        className: m(i({
          variant: s,
          structure: "title"
        }), "text-emphasis line-clamp-1 font-bold leading-5"),
        children: h
      }), c && /* @__PURE__ */ r("p", {
        title: c.toString(),
        className: m(i({
          variant: s,
          structure: "description"
        }), "pt-1"),
        children: c
      })]
    }), s === "SidebarCard" && /* @__PURE__ */ d("a", {
      onClick: e == null ? void 0 : e.onClick,
      target: "_blank",
      rel: "noreferrer",
      href: x,
      "data-testId": e == null ? void 0 : e["data-testId"],
      className: "group relative my-3 flex aspect-video items-center overflow-hidden rounded",
      children: [/* @__PURE__ */ r("div", {
        className: "absolute inset-0 bg-black bg-opacity-50 transition-opacity group-hover:bg-opacity-40"
      }), /* @__PURE__ */ d("svg", {
        className: "text-inverted absolute left-1/2 top-1/2 h-8 w-8 -translate-x-1/2 -translate-y-1/2 transform rounded-full shadow-lg hover:-mt-px",
        viewBox: "0 0 32 32",
        fill: "none",
        xmlns: "http://www.w3.org/2000/svg",
        children: [/* @__PURE__ */ r("path", {
          d: "M16 32C24.8366 32 32 24.8366 32 16C32 7.16344 24.8366 0 16 0C7.16344 0 0 7.16344 0 16C0 24.8366 7.16344 32 16 32Z",
          fill: "white"
        }), /* @__PURE__ */ r("path", {
          d: "M12.1667 8.5L23.8334 16L12.1667 23.5V8.5Z",
          fill: "#111827",
          stroke: "#111827",
          strokeWidth: "1.5",
          strokeLinecap: "round",
          strokeLinejoin: "round"
        })]
      }), /* @__PURE__ */ r("img", {
        alt: "play feature video",
        src: v
      })]
    }), s === "basic" && e && /* @__PURE__ */ r("div", {
      children: /* @__PURE__ */ r(u, {
        color: "secondary",
        href: e == null ? void 0 : e.href,
        className: "mt-10",
        EndIcon: "arrow-right",
        "data-testId": e["data-testId"],
        children: e == null ? void 0 : e.child
      })
    }), s === "SidebarCard" && /* @__PURE__ */ d("div", {
      className: "mt-2 flex items-center justify-between",
      children: [l && /* @__PURE__ */ r(C, {
        href: l.href,
        onClick: e == null ? void 0 : e.onClick,
        target: "_blank",
        rel: "noreferrer",
        className: "text-default text-xs font-medium",
        children: l.text
      }), (e == null ? void 0 : e.child) && /* @__PURE__ */ r("button", {
        className: "text-default hover:text-emphasis p-0 text-xs font-normal",
        color: "minimal",
        "data-testId": e == null ? void 0 : e["data-testId"],
        onClick: e == null ? void 0 : e.onClick,
        children: e == null ? void 0 : e.child
      })]
    })]
  });
}
export {
  g as C
};
