// @calcom/atoms/dist/DatePicker-pErY4gNZ.js
import { an as u, at as g, au as y, av as T, aw as a, ax as M, ay as x } from "./index-6PQ0n_zn.js";
import { jsx as j } from "react/jsx-runtime";
const b = ({
  event: n,
  schedule: e,
  classNames: t
}) => {
  var D, k, d, P, Y;
  const {
    i18n: C
  } = u(), [r, S] = g((o) => [o.month, o.selectedDate], y), [i, p, w] = g((o) => [o.setSelectedDate, o.setMonth, o.setDayCount], y), f = T((D = e == null ? void 0 : e.data) == null ? void 0 : D.slots);
  return /* @__PURE__ */ j(a, {
    customClassNames: {
      datePickerTitle: t == null ? void 0 : t.datePickerTitle,
      datePickerDays: t == null ? void 0 : t.datePickerDays,
      datePickersDates: t == null ? void 0 : t.datePickerDate,
      datePickerDatesActive: t == null ? void 0 : t.datePickerDatesActive,
      datePickerToggle: t == null ? void 0 : t.datePickerToggle
    },
    className: t == null ? void 0 : t.datePickerContainer,
    isPending: e.isPending,
    onChange: (o) => {
      i(o === null ? o : o.format("YYYY-MM-DD"));
    },
    onMonthChange: (o) => {
      p(o.format("YYYY-MM")), i(o.format("YYYY-MM-DD")), w(null);
    },
    includedDates: f,
    locale: C.language,
    browsingDate: r ? M(r) : void 0,
    selected: M(S),
    weekStart: x((P = (d = (k = n == null ? void 0 : n.data) == null ? void 0 : k.users) == null ? void 0 : d[0]) == null ? void 0 : P.weekStart),
    slots: (Y = e == null ? void 0 : e.data) == null ? void 0 : Y.slots
  });
};
export {
  b as DatePicker
};
