// @calcom/atoms/dist/PhoneInput-6AZAVvaQ.js
import Lt, { useState as Rt, useEffect as zt } from "react";
import { aH as Bt, az as Gt, aG as $t } from "./index-6PQ0n_zn.js";
import { jsx as Vt } from "react/jsx-runtime";
import { i as Kt } from "./isSupportedCountry-l4nZ7iA1.js";
var Ut = function(Z) {
  var Q = {};
  function l(u) {
    if (Q[u])
      return Q[u].exports;
    var O = Q[u] = { i: u, l: !1, exports: {} };
    return Z[u].call(O.exports, O, O.exports, l), O.l = !0, O.exports;
  }
  return l.m = Z, l.c = Q, l.d = function(u, O, F) {
    l.o(u, O) || Object.defineProperty(u, O, { enumerable: !0, get: F });
  }, l.r = function(u) {
    typeof Symbol < "u" && Symbol.toStringTag && Object.defineProperty(u, Symbol.toStringTag, { value: "Module" }), Object.defineProperty(u, "__esModule", { value: !0 });
  }, l.t = function(u, O) {
    if (1 & O && (u = l(u)), 8 & O || 4 & O && typeof u == "object" && u && u.__esModule)
      return u;
    var F = /* @__PURE__ */ Object.create(null);
    if (l.r(F), Object.defineProperty(F, "default", { enumerable: !0, value: u }), 2 & O && typeof u != "string")
      for (var ie in u)
        l.d(F, ie, (function(se) {
          return u[se];
        }).bind(null, ie));
    return F;
  }, l.n = function(u) {
    var O = u && u.__esModule ? function() {
      return u.default;
    } : function() {
      return u;
    };
    return l.d(O, "a", O), O;
  }, l.o = function(u, O) {
    return Object.prototype.hasOwnProperty.call(u, O);
  }, l.p = "", l(l.s = 9);
}([function(Z, Q) {
  Z.exports = Lt;
}, function(Z, Q, l) {
  var u;
  /*!
    Copyright (c) 2017 Jed Watson.
    Licensed under the MIT License (MIT), see
    http://jedwatson.github.io/classnames
  */
  (function() {
    var O = {}.hasOwnProperty;
    function F() {
      for (var ie = [], se = 0; se < arguments.length; se++) {
        var te = arguments[se];
        if (te) {
          var Se = typeof te;
          if (Se === "string" || Se === "number")
            ie.push(te);
          else if (Array.isArray(te) && te.length) {
            var be = F.apply(null, te);
            be && ie.push(be);
          } else if (Se === "object")
            for (var ge in te)
              O.call(te, ge) && te[ge] && ie.push(ge);
        }
      }
      return ie.join(" ");
    }
    Z.exports ? (F.default = F, Z.exports = F) : (u = (function() {
      return F;
    }).apply(Q, [])) === void 0 || (Z.exports = u);
  })();
}, function(Z, Q, l) {
  (function(u) {
    var O = /^\s+|\s+$/g, F = /^[-+]0x[0-9a-f]+$/i, ie = /^0b[01]+$/i, se = /^0o[0-7]+$/i, te = parseInt, Se = typeof u == "object" && u && u.Object === Object && u, be = typeof self == "object" && self && self.Object === Object && self, ge = Se || be || Function("return this")(), ve = Object.prototype.toString, Oe = ge.Symbol, Ee = Oe ? Oe.prototype : void 0, _e = Ee ? Ee.toString : void 0;
    function g(j) {
      if (typeof j == "string")
        return j;
      if (R(j))
        return _e ? _e.call(j) : "";
      var b = j + "";
      return b == "0" && 1 / j == -1 / 0 ? "-0" : b;
    }
    function ke(j) {
      var b = typeof j;
      return !!j && (b == "object" || b == "function");
    }
    function R(j) {
      return typeof j == "symbol" || /* @__PURE__ */ function(b) {
        return !!b && typeof b == "object";
      }(j) && ve.call(j) == "[object Symbol]";
    }
    function V(j) {
      return j ? (j = function(b) {
        if (typeof b == "number")
          return b;
        if (R(b))
          return NaN;
        if (ke(b)) {
          var q = typeof b.valueOf == "function" ? b.valueOf() : b;
          b = ke(q) ? q + "" : q;
        }
        if (typeof b != "string")
          return b === 0 ? b : +b;
        b = b.replace(O, "");
        var ue = ie.test(b);
        return ue || se.test(b) ? te(b.slice(2), ue ? 2 : 8) : F.test(b) ? NaN : +b;
      }(j)) === 1 / 0 || j === -1 / 0 ? 17976931348623157e292 * (j < 0 ? -1 : 1) : j == j ? j : 0 : j === 0 ? j : 0;
    }
    Z.exports = function(j, b, q) {
      var ue, G, le, K;
      return j = (ue = j) == null ? "" : g(ue), G = function(Ce) {
        var me = V(Ce), we = me % 1;
        return me == me ? we ? me - we : me : 0;
      }(q), le = 0, K = j.length, G == G && (K !== void 0 && (G = G <= K ? G : K), le !== void 0 && (G = G >= le ? G : le)), q = G, b = g(b), j.slice(q, q + b.length) == b;
    };
  }).call(this, l(3));
}, function(Z, Q) {
  var l;
  l = /* @__PURE__ */ function() {
    return this;
  }();
  try {
    l = l || new Function("return this")();
  } catch {
    typeof window == "object" && (l = window);
  }
  Z.exports = l;
}, function(Z, Q, l) {
  (function(u) {
    var O = /^\[object .+?Constructor\]$/, F = typeof u == "object" && u && u.Object === Object && u, ie = typeof self == "object" && self && self.Object === Object && self, se = F || ie || Function("return this")(), te, Se = Array.prototype, be = Function.prototype, ge = Object.prototype, ve = se["__core-js_shared__"], Oe = (te = /[^.]+$/.exec(ve && ve.keys && ve.keys.IE_PROTO || "")) ? "Symbol(src)_1." + te : "", Ee = be.toString, _e = ge.hasOwnProperty, g = ge.toString, ke = RegExp("^" + Ee.call(_e).replace(/[\\^$.*+?()[\]{}|]/g, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"), R = Se.splice, V = Ce(se, "Map"), j = Ce(Object, "create");
    function b(s) {
      var y = -1, C = s ? s.length : 0;
      for (this.clear(); ++y < C; ) {
        var E = s[y];
        this.set(E[0], E[1]);
      }
    }
    function q(s) {
      var y = -1, C = s ? s.length : 0;
      for (this.clear(); ++y < C; ) {
        var E = s[y];
        this.set(E[0], E[1]);
      }
    }
    function ue(s) {
      var y = -1, C = s ? s.length : 0;
      for (this.clear(); ++y < C; ) {
        var E = s[y];
        this.set(E[0], E[1]);
      }
    }
    function G(s, y) {
      for (var C, E, fe = s.length; fe--; )
        if ((C = s[fe][0]) === (E = y) || C != C && E != E)
          return fe;
      return -1;
    }
    function le(s) {
      return !(!we(s) || (y = s, Oe && Oe in y)) && (function(C) {
        var E = we(C) ? g.call(C) : "";
        return E == "[object Function]" || E == "[object GeneratorFunction]";
      }(s) || function(C) {
        var E = !1;
        if (C != null && typeof C.toString != "function")
          try {
            E = !!(C + "");
          } catch {
          }
        return E;
      }(s) ? ke : O).test(function(C) {
        if (C != null) {
          try {
            return Ee.call(C);
          } catch {
          }
          try {
            return C + "";
          } catch {
          }
        }
        return "";
      }(s));
      var y;
    }
    function K(s, y) {
      var C, E, fe = s.__data__;
      return ((E = typeof (C = y)) == "string" || E == "number" || E == "symbol" || E == "boolean" ? C !== "__proto__" : C === null) ? fe[typeof y == "string" ? "string" : "hash"] : fe.map;
    }
    function Ce(s, y) {
      var C = function(E, fe) {
        return E == null ? void 0 : E[fe];
      }(s, y);
      return le(C) ? C : void 0;
    }
    function me(s, y) {
      if (typeof s != "function" || y && typeof y != "function")
        throw new TypeError("Expected a function");
      var C = function() {
        var E = arguments, fe = y ? y.apply(this, E) : E[0], Ae = C.cache;
        if (Ae.has(fe))
          return Ae.get(fe);
        var m = s.apply(this, E);
        return C.cache = Ae.set(fe, m), m;
      };
      return C.cache = new (me.Cache || ue)(), C;
    }
    function we(s) {
      var y = typeof s;
      return !!s && (y == "object" || y == "function");
    }
    b.prototype.clear = function() {
      this.__data__ = j ? j(null) : {};
    }, b.prototype.delete = function(s) {
      return this.has(s) && delete this.__data__[s];
    }, b.prototype.get = function(s) {
      var y = this.__data__;
      if (j) {
        var C = y[s];
        return C === "__lodash_hash_undefined__" ? void 0 : C;
      }
      return _e.call(y, s) ? y[s] : void 0;
    }, b.prototype.has = function(s) {
      var y = this.__data__;
      return j ? y[s] !== void 0 : _e.call(y, s);
    }, b.prototype.set = function(s, y) {
      return this.__data__[s] = j && y === void 0 ? "__lodash_hash_undefined__" : y, this;
    }, q.prototype.clear = function() {
      this.__data__ = [];
    }, q.prototype.delete = function(s) {
      var y = this.__data__, C = G(y, s);
      return !(C < 0) && (C == y.length - 1 ? y.pop() : R.call(y, C, 1), !0);
    }, q.prototype.get = function(s) {
      var y = this.__data__, C = G(y, s);
      return C < 0 ? void 0 : y[C][1];
    }, q.prototype.has = function(s) {
      return G(this.__data__, s) > -1;
    }, q.prototype.set = function(s, y) {
      var C = this.__data__, E = G(C, s);
      return E < 0 ? C.push([s, y]) : C[E][1] = y, this;
    }, ue.prototype.clear = function() {
      this.__data__ = { hash: new b(), map: new (V || q)(), string: new b() };
    }, ue.prototype.delete = function(s) {
      return K(this, s).delete(s);
    }, ue.prototype.get = function(s) {
      return K(this, s).get(s);
    }, ue.prototype.has = function(s) {
      return K(this, s).has(s);
    }, ue.prototype.set = function(s, y) {
      return K(this, s).set(s, y), this;
    }, me.Cache = ue, Z.exports = me;
  }).call(this, l(3));
}, function(Z, Q, l) {
  (function(u) {
    var O = /^\s+|\s+$/g, F = /^[-+]0x[0-9a-f]+$/i, ie = /^0b[01]+$/i, se = /^0o[0-7]+$/i, te = parseInt, Se = typeof u == "object" && u && u.Object === Object && u, be = typeof self == "object" && self && self.Object === Object && self, ge = Se || be || Function("return this")(), ve = Object.prototype.toString, Oe = Math.max, Ee = Math.min, _e = function() {
      return ge.Date.now();
    };
    function g(R) {
      var V = typeof R;
      return !!R && (V == "object" || V == "function");
    }
    function ke(R) {
      if (typeof R == "number")
        return R;
      if (function(b) {
        return typeof b == "symbol" || /* @__PURE__ */ function(q) {
          return !!q && typeof q == "object";
        }(b) && ve.call(b) == "[object Symbol]";
      }(R))
        return NaN;
      if (g(R)) {
        var V = typeof R.valueOf == "function" ? R.valueOf() : R;
        R = g(V) ? V + "" : V;
      }
      if (typeof R != "string")
        return R === 0 ? R : +R;
      R = R.replace(O, "");
      var j = ie.test(R);
      return j || se.test(R) ? te(R.slice(2), j ? 2 : 8) : F.test(R) ? NaN : +R;
    }
    Z.exports = function(R, V, j) {
      var b, q, ue, G, le, K, Ce = 0, me = !1, we = !1, s = !0;
      if (typeof R != "function")
        throw new TypeError("Expected a function");
      function y(c) {
        var x = b, A = q;
        return b = q = void 0, Ce = c, G = R.apply(A, x);
      }
      function C(c) {
        return Ce = c, le = setTimeout(fe, V), me ? y(c) : G;
      }
      function E(c) {
        var x = c - K;
        return K === void 0 || x >= V || x < 0 || we && c - Ce >= ue;
      }
      function fe() {
        var c = _e();
        if (E(c))
          return Ae(c);
        le = setTimeout(fe, function(x) {
          var A = V - (x - K);
          return we ? Ee(A, ue - (x - Ce)) : A;
        }(c));
      }
      function Ae(c) {
        return le = void 0, s && b ? y(c) : (b = q = void 0, G);
      }
      function m() {
        var c = _e(), x = E(c);
        if (b = arguments, q = this, K = c, x) {
          if (le === void 0)
            return C(K);
          if (we)
            return le = setTimeout(fe, V), y(K);
        }
        return le === void 0 && (le = setTimeout(fe, V)), G;
      }
      return V = ke(V) || 0, g(j) && (me = !!j.leading, ue = (we = "maxWait" in j) ? Oe(ke(j.maxWait) || 0, V) : ue, s = "trailing" in j ? !!j.trailing : s), m.cancel = function() {
        le !== void 0 && clearTimeout(le), Ce = 0, b = K = q = le = void 0;
      }, m.flush = function() {
        return le === void 0 ? G : Ae(_e());
      }, m;
    };
  }).call(this, l(3));
}, function(Z, Q, l) {
  (function(u, O) {
    var F = "[object Arguments]", ie = "[object Map]", se = "[object Object]", te = "[object Set]", Se = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, be = /^\w*$/, ge = /^\./, ve = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, Oe = /\\(\\)?/g, Ee = /^\[object .+?Constructor\]$/, _e = /^(?:0|[1-9]\d*)$/, g = {};
    g["[object Float32Array]"] = g["[object Float64Array]"] = g["[object Int8Array]"] = g["[object Int16Array]"] = g["[object Int32Array]"] = g["[object Uint8Array]"] = g["[object Uint8ClampedArray]"] = g["[object Uint16Array]"] = g["[object Uint32Array]"] = !0, g[F] = g["[object Array]"] = g["[object ArrayBuffer]"] = g["[object Boolean]"] = g["[object DataView]"] = g["[object Date]"] = g["[object Error]"] = g["[object Function]"] = g[ie] = g["[object Number]"] = g[se] = g["[object RegExp]"] = g[te] = g["[object String]"] = g["[object WeakMap]"] = !1;
    var ke = typeof u == "object" && u && u.Object === Object && u, R = typeof self == "object" && self && self.Object === Object && self, V = ke || R || Function("return this")(), j = Q && !Q.nodeType && Q, b = j && typeof O == "object" && O && !O.nodeType && O, q = b && b.exports === j && ke.process, ue = function() {
      try {
        return q && q.binding("util");
      } catch {
      }
    }(), G = ue && ue.isTypedArray;
    function le(e, r, a, i) {
      var _ = -1, w = e ? e.length : 0;
      for (i && w && (a = e[++_]); ++_ < w; )
        a = r(a, e[_], _, e);
      return a;
    }
    function K(e, r) {
      for (var a = -1, i = e ? e.length : 0; ++a < i; )
        if (r(e[a], a, e))
          return !0;
      return !1;
    }
    function Ce(e, r, a, i, _) {
      return _(e, function(w, I, W) {
        a = i ? (i = !1, w) : r(a, w, I, W);
      }), a;
    }
    function me(e) {
      var r = !1;
      if (e != null && typeof e.toString != "function")
        try {
          r = !!(e + "");
        } catch {
        }
      return r;
    }
    function we(e) {
      var r = -1, a = Array(e.size);
      return e.forEach(function(i, _) {
        a[++r] = [_, i];
      }), a;
    }
    function s(e) {
      var r = -1, a = Array(e.size);
      return e.forEach(function(i) {
        a[++r] = i;
      }), a;
    }
    var y, C, E, fe = Array.prototype, Ae = Function.prototype, m = Object.prototype, c = V["__core-js_shared__"], x = (y = /[^.]+$/.exec(c && c.keys && c.keys.IE_PROTO || "")) ? "Symbol(src)_1." + y : "", A = Ae.toString, f = m.hasOwnProperty, t = m.toString, z = RegExp("^" + A.call(f).replace(/[\\^$.*+?()[\]{}|]/g, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"), M = V.Symbol, L = V.Uint8Array, re = m.propertyIsEnumerable, Y = fe.splice, ne = (C = Object.keys, E = Object, function(e) {
      return C(E(e));
    }), ce = qe(V, "DataView"), de = qe(V, "Map"), ae = qe(V, "Promise"), n = qe(V, "Set"), o = qe(V, "WeakMap"), d = qe(Object, "create"), k = $e(ce), N = $e(de), T = $e(ae), D = $e(n), U = $e(o), p = M ? M.prototype : void 0, S = p ? p.valueOf : void 0, P = p ? p.toString : void 0;
    function h(e) {
      var r = -1, a = e ? e.length : 0;
      for (this.clear(); ++r < a; ) {
        var i = e[r];
        this.set(i[0], i[1]);
      }
    }
    function v(e) {
      var r = -1, a = e ? e.length : 0;
      for (this.clear(); ++r < a; ) {
        var i = e[r];
        this.set(i[0], i[1]);
      }
    }
    function X(e) {
      var r = -1, a = e ? e.length : 0;
      for (this.clear(); ++r < a; ) {
        var i = e[r];
        this.set(i[0], i[1]);
      }
    }
    function je(e) {
      var r = -1, a = e ? e.length : 0;
      for (this.__data__ = new X(); ++r < a; )
        this.add(e[r]);
    }
    function ye(e) {
      this.__data__ = new v(e);
    }
    function Te(e, r) {
      var a = Le(e) || xt(e) ? function(I, W) {
        for (var ee = -1, $ = Array(I); ++ee < I; )
          $[ee] = W(ee);
        return $;
      }(e.length, String) : [], i = a.length, _ = !!i;
      for (var w in e)
        !r && !f.call(e, w) || _ && (w == "length" || _t(w, i)) || a.push(w);
      return a;
    }
    function Pe(e, r) {
      for (var a = e.length; a--; )
        if (jt(e[a][0], r))
          return a;
      return -1;
    }
    h.prototype.clear = function() {
      this.__data__ = d ? d(null) : {};
    }, h.prototype.delete = function(e) {
      return this.has(e) && delete this.__data__[e];
    }, h.prototype.get = function(e) {
      var r = this.__data__;
      if (d) {
        var a = r[e];
        return a === "__lodash_hash_undefined__" ? void 0 : a;
      }
      return f.call(r, e) ? r[e] : void 0;
    }, h.prototype.has = function(e) {
      var r = this.__data__;
      return d ? r[e] !== void 0 : f.call(r, e);
    }, h.prototype.set = function(e, r) {
      return this.__data__[e] = d && r === void 0 ? "__lodash_hash_undefined__" : r, this;
    }, v.prototype.clear = function() {
      this.__data__ = [];
    }, v.prototype.delete = function(e) {
      var r = this.__data__, a = Pe(r, e);
      return !(a < 0) && (a == r.length - 1 ? r.pop() : Y.call(r, a, 1), !0);
    }, v.prototype.get = function(e) {
      var r = this.__data__, a = Pe(r, e);
      return a < 0 ? void 0 : r[a][1];
    }, v.prototype.has = function(e) {
      return Pe(this.__data__, e) > -1;
    }, v.prototype.set = function(e, r) {
      var a = this.__data__, i = Pe(a, e);
      return i < 0 ? a.push([e, r]) : a[i][1] = r, this;
    }, X.prototype.clear = function() {
      this.__data__ = { hash: new h(), map: new (de || v)(), string: new h() };
    }, X.prototype.delete = function(e) {
      return Xe(this, e).delete(e);
    }, X.prototype.get = function(e) {
      return Xe(this, e).get(e);
    }, X.prototype.has = function(e) {
      return Xe(this, e).has(e);
    }, X.prototype.set = function(e, r) {
      return Xe(this, e).set(e, r), this;
    }, je.prototype.add = je.prototype.push = function(e) {
      return this.__data__.set(e, "__lodash_hash_undefined__"), this;
    }, je.prototype.has = function(e) {
      return this.__data__.has(e);
    }, ye.prototype.clear = function() {
      this.__data__ = new v();
    }, ye.prototype.delete = function(e) {
      return this.__data__.delete(e);
    }, ye.prototype.get = function(e) {
      return this.__data__.get(e);
    }, ye.prototype.has = function(e) {
      return this.__data__.has(e);
    }, ye.prototype.set = function(e, r) {
      var a = this.__data__;
      if (a instanceof v) {
        var i = a.__data__;
        if (!de || i.length < 199)
          return i.push([e, r]), this;
        a = this.__data__ = new X(i);
      }
      return a.set(e, r), this;
    };
    var Ge, pe = (Ge = function(e, r) {
      return e && Ue(e, r, at);
    }, function(e, r) {
      if (e == null)
        return e;
      if (!ht(e))
        return Ge(e, r);
      for (var a = e.length, i = -1, _ = Object(e); ++i < a && r(_[i], i, _) !== !1; )
        ;
      return e;
    }), Ue = /* @__PURE__ */ function(e) {
      return function(r, a, i) {
        for (var _ = -1, w = Object(r), I = i(r), W = I.length; W--; ) {
          var ee = I[e ? W : ++_];
          if (a(w[ee], ee, w) === !1)
            break;
        }
        return r;
      };
    }();
    function We(e, r) {
      for (var a = 0, i = (r = et(r, e) ? [r] : vt(r)).length; e != null && a < i; )
        e = e[tt(r[a++])];
      return a && a == i ? e : void 0;
    }
    function dt(e, r) {
      return e != null && r in Object(e);
    }
    function Je(e, r, a, i, _) {
      return e === r || (e == null || r == null || !rt(e) && !nt(r) ? e != e && r != r : function(w, I, W, ee, $, B) {
        var Ie = Le(w), De = Le(I), oe = "[object Array]", he = "[object Array]";
        Ie || (oe = (oe = Me(w)) == F ? se : oe), De || (he = (he = Me(I)) == F ? se : he);
        var xe = oe == se && !me(w), Ve = he == se && !me(I), Re = oe == he;
        if (Re && !xe)
          return B || (B = new ye()), Ie || At(w) ? Ct(w, I, W, ee, $, B) : function(J, H, bt, Ke, ot, Ne, Fe) {
            switch (bt) {
              case "[object DataView]":
                if (J.byteLength != H.byteLength || J.byteOffset != H.byteOffset)
                  return !1;
                J = J.buffer, H = H.buffer;
              case "[object ArrayBuffer]":
                return !(J.byteLength != H.byteLength || !Ke(new L(J), new L(H)));
              case "[object Boolean]":
              case "[object Date]":
              case "[object Number]":
                return jt(+J, +H);
              case "[object Error]":
                return J.name == H.name && J.message == H.message;
              case "[object RegExp]":
              case "[object String]":
                return J == H + "";
              case ie:
                var ze = we;
              case te:
                var Ye = 2 & Ne;
                if (ze || (ze = s), J.size != H.size && !Ye)
                  return !1;
                var it = Fe.get(J);
                if (it)
                  return it == H;
                Ne |= 1, Fe.set(J, H);
                var He = Ct(ze(J), ze(H), Ke, ot, Ne, Fe);
                return Fe.delete(J), He;
              case "[object Symbol]":
                if (S)
                  return S.call(J) == S.call(H);
            }
            return !1;
          }(w, I, oe, W, ee, $, B);
        if (!(2 & $)) {
          var Qe = xe && f.call(w, "__wrapped__"), Ot = Ve && f.call(I, "__wrapped__");
          if (Qe || Ot) {
            var Ft = Qe ? w.value() : w, Mt = Ot ? I.value() : I;
            return B || (B = new ye()), W(Ft, Mt, ee, $, B);
          }
        }
        return Re ? (B || (B = new ye()), function(J, H, bt, Ke, ot, Ne) {
          var Fe = 2 & ot, ze = at(J), Ye = ze.length, it = at(H).length;
          if (Ye != it && !Fe)
            return !1;
          for (var He = Ye; He--; ) {
            var Be = ze[He];
            if (!(Fe ? Be in H : f.call(H, Be)))
              return !1;
          }
          var Et = Ne.get(J);
          if (Et && Ne.get(H))
            return Et == H;
          var ut = !0;
          Ne.set(J, H), Ne.set(H, J);
          for (var gt = Fe; ++He < Ye; ) {
            Be = ze[He];
            var ct = J[Be], st = H[Be];
            if (Ke)
              var kt = Fe ? Ke(st, ct, Be, H, J, Ne) : Ke(ct, st, Be, J, H, Ne);
            if (!(kt === void 0 ? ct === st || bt(ct, st, Ke, ot, Ne) : kt)) {
              ut = !1;
              break;
            }
            gt || (gt = Be == "constructor");
          }
          if (ut && !gt) {
            var lt = J.constructor, ft = H.constructor;
            lt == ft || !("constructor" in J) || !("constructor" in H) || typeof lt == "function" && lt instanceof lt && typeof ft == "function" && ft instanceof ft || (ut = !1);
          }
          return Ne.delete(J), Ne.delete(H), ut;
        }(w, I, W, ee, $, B)) : !1;
      }(e, r, Je, a, i, _));
    }
    function Ze(e) {
      return !(!rt(e) || function(r) {
        return !!x && x in r;
      }(e)) && (Nt(e) || me(e) ? z : Ee).test($e(e));
    }
    function Tt(e) {
      return typeof e == "function" ? e : e == null ? Pt : typeof e == "object" ? Le(e) ? function(i, _) {
        return et(i) && wt(_) ? St(tt(i), _) : function(w) {
          var I = function(W, ee, $) {
            var B = W == null ? void 0 : We(W, ee);
            return B === void 0 ? $ : B;
          }(w, i);
          return I === void 0 && I === _ ? function(W, ee) {
            return W != null && function($, B, Ie) {
              B = et(B, $) ? [B] : vt(B);
              for (var De, oe = -1, he = B.length; ++oe < he; ) {
                var xe = tt(B[oe]);
                if (!(De = $ != null && Ie($, xe)))
                  break;
                $ = $[xe];
              }
              return De || !!(he = $ ? $.length : 0) && mt(he) && _t(xe, he) && (Le($) || xt($));
            }(W, ee, dt);
          }(w, i) : Je(_, I, void 0, 3);
        };
      }(e[0], e[1]) : function(i) {
        var _ = function(w) {
          for (var I = at(w), W = I.length; W--; ) {
            var ee = I[W], $ = w[ee];
            I[W] = [ee, $, wt($)];
          }
          return I;
        }(i);
        return _.length == 1 && _[0][2] ? St(_[0][0], _[0][1]) : function(w) {
          return w === i || function(I, W, ee, $) {
            var B = ee.length, Ie = B, De = !$;
            if (I == null)
              return !Ie;
            for (I = Object(I); B--; ) {
              var oe = ee[B];
              if (De && oe[2] ? oe[1] !== I[oe[0]] : !(oe[0] in I))
                return !1;
            }
            for (; ++B < Ie; ) {
              var he = (oe = ee[B])[0], xe = I[he], Ve = oe[1];
              if (De && oe[2]) {
                if (xe === void 0 && !(he in I))
                  return !1;
              } else {
                var Re = new ye();
                if ($)
                  var Qe = $(xe, Ve, he, I, W, Re);
                if (!(Qe === void 0 ? Je(Ve, xe, $, 3, Re) : Qe))
                  return !1;
              }
            }
            return !0;
          }(w, i, _);
        };
      }(e) : et(r = e) ? (a = tt(r), function(i) {
        return i == null ? void 0 : i[a];
      }) : /* @__PURE__ */ function(i) {
        return function(_) {
          return We(_, i);
        };
      }(r);
      var r, a;
    }
    function It(e) {
      if (a = (r = e) && r.constructor, i = typeof a == "function" && a.prototype || m, r !== i)
        return ne(e);
      var r, a, i, _ = [];
      for (var w in Object(e))
        f.call(e, w) && w != "constructor" && _.push(w);
      return _;
    }
    function vt(e) {
      return Le(e) ? e : Dt(e);
    }
    function Ct(e, r, a, i, _, w) {
      var I = 2 & _, W = e.length, ee = r.length;
      if (W != ee && !(I && ee > W))
        return !1;
      var $ = w.get(e);
      if ($ && w.get(r))
        return $ == r;
      var B = -1, Ie = !0, De = 1 & _ ? new je() : void 0;
      for (w.set(e, r), w.set(r, e); ++B < W; ) {
        var oe = e[B], he = r[B];
        if (i)
          var xe = I ? i(he, oe, B, r, e, w) : i(oe, he, B, e, r, w);
        if (xe !== void 0) {
          if (xe)
            continue;
          Ie = !1;
          break;
        }
        if (De) {
          if (!K(r, function(Ve, Re) {
            if (!De.has(Re) && (oe === Ve || a(oe, Ve, i, _, w)))
              return De.add(Re);
          })) {
            Ie = !1;
            break;
          }
        } else if (oe !== he && !a(oe, he, i, _, w)) {
          Ie = !1;
          break;
        }
      }
      return w.delete(e), w.delete(r), Ie;
    }
    function Xe(e, r) {
      var a, i, _ = e.__data__;
      return ((i = typeof (a = r)) == "string" || i == "number" || i == "symbol" || i == "boolean" ? a !== "__proto__" : a === null) ? _[typeof r == "string" ? "string" : "hash"] : _.map;
    }
    function qe(e, r) {
      var a = function(i, _) {
        return i == null ? void 0 : i[_];
      }(e, r);
      return Ze(a) ? a : void 0;
    }
    var Me = function(e) {
      return t.call(e);
    };
    function _t(e, r) {
      return !!(r = r ?? 9007199254740991) && (typeof e == "number" || _e.test(e)) && e > -1 && e % 1 == 0 && e < r;
    }
    function et(e, r) {
      if (Le(e))
        return !1;
      var a = typeof e;
      return !(a != "number" && a != "symbol" && a != "boolean" && e != null && !yt(e)) || be.test(e) || !Se.test(e) || r != null && e in Object(r);
    }
    function wt(e) {
      return e == e && !rt(e);
    }
    function St(e, r) {
      return function(a) {
        return a != null && a[e] === r && (r !== void 0 || e in Object(a));
      };
    }
    (ce && Me(new ce(new ArrayBuffer(1))) != "[object DataView]" || de && Me(new de()) != ie || ae && Me(ae.resolve()) != "[object Promise]" || n && Me(new n()) != te || o && Me(new o()) != "[object WeakMap]") && (Me = function(e) {
      var r = t.call(e), a = r == se ? e.constructor : void 0, i = a ? $e(a) : void 0;
      if (i)
        switch (i) {
          case k:
            return "[object DataView]";
          case N:
            return ie;
          case T:
            return "[object Promise]";
          case D:
            return te;
          case U:
            return "[object WeakMap]";
        }
      return r;
    });
    var Dt = pt(function(e) {
      var r;
      e = (r = e) == null ? "" : function(i) {
        if (typeof i == "string")
          return i;
        if (yt(i))
          return P ? P.call(i) : "";
        var _ = i + "";
        return _ == "0" && 1 / i == -1 / 0 ? "-0" : _;
      }(r);
      var a = [];
      return ge.test(e) && a.push(""), e.replace(ve, function(i, _, w, I) {
        a.push(w ? I.replace(Oe, "$1") : _ || i);
      }), a;
    });
    function tt(e) {
      if (typeof e == "string" || yt(e))
        return e;
      var r = e + "";
      return r == "0" && 1 / e == -1 / 0 ? "-0" : r;
    }
    function $e(e) {
      if (e != null) {
        try {
          return A.call(e);
        } catch {
        }
        try {
          return e + "";
        } catch {
        }
      }
      return "";
    }
    function pt(e, r) {
      if (typeof e != "function" || r && typeof r != "function")
        throw new TypeError("Expected a function");
      var a = function() {
        var i = arguments, _ = r ? r.apply(this, i) : i[0], w = a.cache;
        if (w.has(_))
          return w.get(_);
        var I = e.apply(this, i);
        return a.cache = w.set(_, I), I;
      };
      return a.cache = new (pt.Cache || X)(), a;
    }
    function jt(e, r) {
      return e === r || e != e && r != r;
    }
    function xt(e) {
      return function(r) {
        return nt(r) && ht(r);
      }(e) && f.call(e, "callee") && (!re.call(e, "callee") || t.call(e) == F);
    }
    pt.Cache = X;
    var Le = Array.isArray;
    function ht(e) {
      return e != null && mt(e.length) && !Nt(e);
    }
    function Nt(e) {
      var r = rt(e) ? t.call(e) : "";
      return r == "[object Function]" || r == "[object GeneratorFunction]";
    }
    function mt(e) {
      return typeof e == "number" && e > -1 && e % 1 == 0 && e <= 9007199254740991;
    }
    function rt(e) {
      var r = typeof e;
      return !!e && (r == "object" || r == "function");
    }
    function nt(e) {
      return !!e && typeof e == "object";
    }
    function yt(e) {
      return typeof e == "symbol" || nt(e) && t.call(e) == "[object Symbol]";
    }
    var At = G ? /* @__PURE__ */ function(e) {
      return function(r) {
        return e(r);
      };
    }(G) : function(e) {
      return nt(e) && mt(e.length) && !!g[t.call(e)];
    };
    function at(e) {
      return ht(e) ? Te(e) : It(e);
    }
    function Pt(e) {
      return e;
    }
    O.exports = function(e, r, a) {
      var i = Le(e) ? le : Ce, _ = arguments.length < 3;
      return i(e, Tt(r), a, _, pe);
    };
  }).call(this, l(3), l(7)(Z));
}, function(Z, Q) {
  Z.exports = function(l) {
    return l.webpackPolyfill || (l.deprecate = function() {
    }, l.paths = [], l.children || (l.children = []), Object.defineProperty(l, "loaded", { enumerable: !0, get: function() {
      return l.l;
    } }), Object.defineProperty(l, "id", { enumerable: !0, get: function() {
      return l.i;
    } }), l.webpackPolyfill = 1), l;
  };
}, function(Z, Q) {
  String.prototype.padEnd || (String.prototype.padEnd = function(l, u) {
    return l >>= 0, u = String(u !== void 0 ? u : " "), this.length > l ? String(this) : ((l -= this.length) > u.length && (u += u.repeat(l / u.length)), String(this) + u.slice(0, l));
  });
}, function(Z, Q, l) {
  function u(m, c, x) {
    return c in m ? Object.defineProperty(m, c, { value: x, enumerable: !0, configurable: !0, writable: !0 }) : m[c] = x, m;
  }
  function O(m) {
    if (Symbol.iterator in Object(m) || Object.prototype.toString.call(m) === "[object Arguments]")
      return Array.from(m);
  }
  function F(m) {
    return function(c) {
      if (Array.isArray(c)) {
        for (var x = 0, A = new Array(c.length); x < c.length; x++)
          A[x] = c[x];
        return A;
      }
    }(m) || O(m) || function() {
      throw new TypeError("Invalid attempt to spread non-iterable instance");
    }();
  }
  function ie(m) {
    if (Array.isArray(m))
      return m;
  }
  function se() {
    throw new TypeError("Invalid attempt to destructure non-iterable instance");
  }
  function te(m, c) {
    if (!(m instanceof c))
      throw new TypeError("Cannot call a class as a function");
  }
  function Se(m, c) {
    for (var x = 0; x < c.length; x++) {
      var A = c[x];
      A.enumerable = A.enumerable || !1, A.configurable = !0, "value" in A && (A.writable = !0), Object.defineProperty(m, A.key, A);
    }
  }
  function be(m) {
    return (be = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(c) {
      return typeof c;
    } : function(c) {
      return c && typeof Symbol == "function" && c.constructor === Symbol && c !== Symbol.prototype ? "symbol" : typeof c;
    })(m);
  }
  function ge(m) {
    return (ge = typeof Symbol == "function" && be(Symbol.iterator) === "symbol" ? function(c) {
      return be(c);
    } : function(c) {
      return c && typeof Symbol == "function" && c.constructor === Symbol && c !== Symbol.prototype ? "symbol" : be(c);
    })(m);
  }
  function ve(m) {
    if (m === void 0)
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    return m;
  }
  function Oe(m) {
    return (Oe = Object.setPrototypeOf ? Object.getPrototypeOf : function(c) {
      return c.__proto__ || Object.getPrototypeOf(c);
    })(m);
  }
  function Ee(m, c) {
    return (Ee = Object.setPrototypeOf || function(x, A) {
      return x.__proto__ = A, x;
    })(m, c);
  }
  l.r(Q);
  var _e = l(0), g = l.n(_e), ke = l(5), R = l.n(ke), V = l(4), j = l.n(V), b = l(6), q = l.n(b), ue = l(2), G = l.n(ue), le = l(1), K = l.n(le);
  l(8);
  function Ce(m, c) {
    return ie(m) || function(x, A) {
      var f = [], t = !0, z = !1, M = void 0;
      try {
        for (var L, re = x[Symbol.iterator](); !(t = (L = re.next()).done) && (f.push(L.value), !A || f.length !== A); t = !0)
          ;
      } catch (Y) {
        z = !0, M = Y;
      } finally {
        try {
          t || re.return == null || re.return();
        } finally {
          if (z)
            throw M;
        }
      }
      return f;
    }(m, c) || se();
  }
  var me = [["Afghanistan", ["asia"], "af", "93"], ["Albania", ["europe"], "al", "355"], ["Algeria", ["africa", "north-africa"], "dz", "213"], ["Andorra", ["europe"], "ad", "376"], ["Angola", ["africa"], "ao", "244"], ["Antigua and Barbuda", ["america", "carribean"], "ag", "1268"], ["Argentina", ["america", "south-america"], "ar", "54", "(..) ........", 0, ["11", "221", "223", "261", "264", "2652", "280", "2905", "291", "2920", "2966", "299", "341", "342", "343", "351", "376", "379", "381", "3833", "385", "387", "388"]], ["Armenia", ["asia", "ex-ussr"], "am", "374", ".. ......"], ["Aruba", ["america", "carribean"], "aw", "297"], ["Australia", ["oceania"], "au", "61", "(..) .... ....", 0, ["2", "3", "4", "7", "8", "02", "03", "04", "07", "08"]], ["Austria", ["europe", "eu-union"], "at", "43"], ["Azerbaijan", ["asia", "ex-ussr"], "az", "994", "(..) ... .. .."], ["Bahamas", ["america", "carribean"], "bs", "1242"], ["Bahrain", ["middle-east"], "bh", "973"], ["Bangladesh", ["asia"], "bd", "880"], ["Barbados", ["america", "carribean"], "bb", "1246"], ["Belarus", ["europe", "ex-ussr"], "by", "375", "(..) ... .. .."], ["Belgium", ["europe", "eu-union"], "be", "32", "... .. .. .."], ["Belize", ["america", "central-america"], "bz", "501"], ["Benin", ["africa"], "bj", "229"], ["Bhutan", ["asia"], "bt", "975"], ["Bolivia", ["america", "south-america"], "bo", "591"], ["Bosnia and Herzegovina", ["europe", "ex-yugos"], "ba", "387"], ["Botswana", ["africa"], "bw", "267"], ["Brazil", ["america", "south-america"], "br", "55", "(..) ........."], ["British Indian Ocean Territory", ["asia"], "io", "246"], ["Brunei", ["asia"], "bn", "673"], ["Bulgaria", ["europe", "eu-union"], "bg", "359"], ["Burkina Faso", ["africa"], "bf", "226"], ["Burundi", ["africa"], "bi", "257"], ["Cambodia", ["asia"], "kh", "855"], ["Cameroon", ["africa"], "cm", "237"], ["Canada", ["america", "north-america"], "ca", "1", "(...) ...-....", 1, ["204", "226", "236", "249", "250", "289", "306", "343", "365", "387", "403", "416", "418", "431", "437", "438", "450", "506", "514", "519", "548", "579", "581", "587", "604", "613", "639", "647", "672", "705", "709", "742", "778", "780", "782", "807", "819", "825", "867", "873", "902", "905"]], ["Cape Verde", ["africa"], "cv", "238"], ["Caribbean Netherlands", ["america", "carribean"], "bq", "599", "", 1], ["Central African Republic", ["africa"], "cf", "236"], ["Chad", ["africa"], "td", "235"], ["Chile", ["america", "south-america"], "cl", "56"], ["China", ["asia"], "cn", "86", "..-........."], ["Colombia", ["america", "south-america"], "co", "57", "... ... ...."], ["Comoros", ["africa"], "km", "269"], ["Congo", ["africa"], "cd", "243"], ["Congo", ["africa"], "cg", "242"], ["Costa Rica", ["america", "central-america"], "cr", "506", "....-...."], ["Côte d’Ivoire", ["africa"], "ci", "225", ".. .. .. .."], ["Croatia", ["europe", "eu-union", "ex-yugos"], "hr", "385"], ["Cuba", ["america", "carribean"], "cu", "53"], ["Curaçao", ["america", "carribean"], "cw", "599", "", 0], ["Cyprus", ["europe", "eu-union"], "cy", "357", ".. ......"], ["Czech Republic", ["europe", "eu-union"], "cz", "420", "... ... ..."], ["Denmark", ["europe", "eu-union", "baltic"], "dk", "45", ".. .. .. .."], ["Djibouti", ["africa"], "dj", "253"], ["Dominica", ["america", "carribean"], "dm", "1767"], ["Dominican Republic", ["america", "carribean"], "do", "1", "", 2, ["809", "829", "849"]], ["Ecuador", ["america", "south-america"], "ec", "593"], ["Egypt", ["africa", "north-africa"], "eg", "20"], ["El Salvador", ["america", "central-america"], "sv", "503", "....-...."], ["Equatorial Guinea", ["africa"], "gq", "240"], ["Eritrea", ["africa"], "er", "291"], ["Estonia", ["europe", "eu-union", "ex-ussr", "baltic"], "ee", "372", ".... ......"], ["Ethiopia", ["africa"], "et", "251"], ["Fiji", ["oceania"], "fj", "679"], ["Finland", ["europe", "eu-union", "baltic"], "fi", "358", ".. ... .. .."], ["France", ["europe", "eu-union"], "fr", "33", ". .. .. .. .."], ["French Guiana", ["america", "south-america"], "gf", "594"], ["French Polynesia", ["oceania"], "pf", "689"], ["Gabon", ["africa"], "ga", "241"], ["Gambia", ["africa"], "gm", "220"], ["Georgia", ["asia", "ex-ussr"], "ge", "995"], ["Germany", ["europe", "eu-union", "baltic"], "de", "49", ".... ........"], ["Ghana", ["africa"], "gh", "233"], ["Greece", ["europe", "eu-union"], "gr", "30"], ["Grenada", ["america", "carribean"], "gd", "1473"], ["Guadeloupe", ["america", "carribean"], "gp", "590", "", 0], ["Guam", ["oceania"], "gu", "1671"], ["Guatemala", ["america", "central-america"], "gt", "502", "....-...."], ["Guinea", ["africa"], "gn", "224"], ["Guinea-Bissau", ["africa"], "gw", "245"], ["Guyana", ["america", "south-america"], "gy", "592"], ["Haiti", ["america", "carribean"], "ht", "509", "....-...."], ["Honduras", ["america", "central-america"], "hn", "504"], ["Hong Kong", ["asia"], "hk", "852", ".... ...."], ["Hungary", ["europe", "eu-union"], "hu", "36"], ["Iceland", ["europe"], "is", "354", "... ...."], ["India", ["asia"], "in", "91", ".....-....."], ["Indonesia", ["asia"], "id", "62"], ["Iran", ["middle-east"], "ir", "98", "... ... ...."], ["Iraq", ["middle-east"], "iq", "964"], ["Ireland", ["europe", "eu-union"], "ie", "353", ".. ......."], ["Israel", ["middle-east"], "il", "972", "... ... ...."], ["Italy", ["europe", "eu-union"], "it", "39", "... .......", 0], ["Jamaica", ["america", "carribean"], "jm", "1876"], ["Japan", ["asia"], "jp", "81", ".. .... ...."], ["Jordan", ["middle-east"], "jo", "962"], ["Kazakhstan", ["asia", "ex-ussr"], "kz", "7", "... ...-..-..", 1, ["310", "311", "312", "313", "315", "318", "321", "324", "325", "326", "327", "336", "7172", "73622"]], ["Kenya", ["africa"], "ke", "254"], ["Kiribati", ["oceania"], "ki", "686"], ["Kosovo", ["europe", "ex-yugos"], "xk", "383"], ["Kuwait", ["middle-east"], "kw", "965"], ["Kyrgyzstan", ["asia", "ex-ussr"], "kg", "996", "... ... ..."], ["Laos", ["asia"], "la", "856"], ["Latvia", ["europe", "eu-union", "ex-ussr", "baltic"], "lv", "371", ".. ... ..."], ["Lebanon", ["middle-east"], "lb", "961"], ["Lesotho", ["africa"], "ls", "266"], ["Liberia", ["africa"], "lr", "231"], ["Libya", ["africa", "north-africa"], "ly", "218"], ["Liechtenstein", ["europe"], "li", "423"], ["Lithuania", ["europe", "eu-union", "ex-ussr", "baltic"], "lt", "370"], ["Luxembourg", ["europe", "eu-union"], "lu", "352"], ["Macau", ["asia"], "mo", "853"], ["Macedonia", ["europe", "ex-yugos"], "mk", "389"], ["Madagascar", ["africa"], "mg", "261"], ["Malawi", ["africa"], "mw", "265"], ["Malaysia", ["asia"], "my", "60", "..-....-...."], ["Maldives", ["asia"], "mv", "960"], ["Mali", ["africa"], "ml", "223"], ["Malta", ["europe", "eu-union"], "mt", "356"], ["Marshall Islands", ["oceania"], "mh", "692"], ["Martinique", ["america", "carribean"], "mq", "596"], ["Mauritania", ["africa"], "mr", "222"], ["Mauritius", ["africa"], "mu", "230"], ["Mexico", ["america", "central-america"], "mx", "52", "... ... ....", 0, ["55", "81", "33", "656", "664", "998", "774", "229"]], ["Micronesia", ["oceania"], "fm", "691"], ["Moldova", ["europe"], "md", "373", "(..) ..-..-.."], ["Monaco", ["europe"], "mc", "377"], ["Mongolia", ["asia"], "mn", "976"], ["Montenegro", ["europe", "ex-yugos"], "me", "382"], ["Morocco", ["africa", "north-africa"], "ma", "212"], ["Mozambique", ["africa"], "mz", "258"], ["Myanmar", ["asia"], "mm", "95"], ["Namibia", ["africa"], "na", "264"], ["Nauru", ["africa"], "nr", "674"], ["Nepal", ["asia"], "np", "977"], ["Netherlands", ["europe", "eu-union"], "nl", "31", ".. ........"], ["New Caledonia", ["oceania"], "nc", "687"], ["New Zealand", ["oceania"], "nz", "64", "...-...-...."], ["Nicaragua", ["america", "central-america"], "ni", "505"], ["Niger", ["africa"], "ne", "227"], ["Nigeria", ["africa"], "ng", "234"], ["North Korea", ["asia"], "kp", "850"], ["Norway", ["europe", "baltic"], "no", "47", "... .. ..."], ["Oman", ["middle-east"], "om", "968"], ["Pakistan", ["asia"], "pk", "92", "...-......."], ["Palau", ["oceania"], "pw", "680"], ["Palestine", ["middle-east"], "ps", "970"], ["Panama", ["america", "central-america"], "pa", "507"], ["Papua New Guinea", ["oceania"], "pg", "675"], ["Paraguay", ["america", "south-america"], "py", "595"], ["Peru", ["america", "south-america"], "pe", "51"], ["Philippines", ["asia"], "ph", "63", ".... ......."], ["Poland", ["europe", "eu-union", "baltic"], "pl", "48", "...-...-..."], ["Portugal", ["europe", "eu-union"], "pt", "351"], ["Puerto Rico", ["america", "carribean"], "pr", "1", "", 3, ["787", "939"]], ["Qatar", ["middle-east"], "qa", "974"], ["Réunion", ["africa"], "re", "262"], ["Romania", ["europe", "eu-union"], "ro", "40"], ["Russia", ["europe", "asia", "ex-ussr", "baltic"], "ru", "7", "(...) ...-..-..", 0], ["Rwanda", ["africa"], "rw", "250"], ["Saint Kitts and Nevis", ["america", "carribean"], "kn", "1869"], ["Saint Lucia", ["america", "carribean"], "lc", "1758"], ["Saint Vincent and the Grenadines", ["america", "carribean"], "vc", "1784"], ["Samoa", ["oceania"], "ws", "685"], ["San Marino", ["europe"], "sm", "378"], ["São Tomé and Príncipe", ["africa"], "st", "239"], ["Saudi Arabia", ["middle-east"], "sa", "966"], ["Senegal", ["africa"], "sn", "221"], ["Serbia", ["europe", "ex-yugos"], "rs", "381"], ["Seychelles", ["africa"], "sc", "248"], ["Sierra Leone", ["africa"], "sl", "232"], ["Singapore", ["asia"], "sg", "65", "....-...."], ["Slovakia", ["europe", "eu-union"], "sk", "421"], ["Slovenia", ["europe", "eu-union", "ex-yugos"], "si", "386"], ["Solomon Islands", ["oceania"], "sb", "677"], ["Somalia", ["africa"], "so", "252"], ["South Africa", ["africa"], "za", "27"], ["South Korea", ["asia"], "kr", "82", "... .... ...."], ["South Sudan", ["africa", "north-africa"], "ss", "211"], ["Spain", ["europe", "eu-union"], "es", "34", "... ... ..."], ["Sri Lanka", ["asia"], "lk", "94"], ["Sudan", ["africa"], "sd", "249"], ["Suriname", ["america", "south-america"], "sr", "597"], ["Swaziland", ["africa"], "sz", "268"], ["Sweden", ["europe", "eu-union", "baltic"], "se", "46", "(...) ...-..."], ["Switzerland", ["europe"], "ch", "41", ".. ... .. .."], ["Syria", ["middle-east"], "sy", "963"], ["Taiwan", ["asia"], "tw", "886"], ["Tajikistan", ["asia", "ex-ussr"], "tj", "992"], ["Tanzania", ["africa"], "tz", "255"], ["Thailand", ["asia"], "th", "66"], ["Timor-Leste", ["asia"], "tl", "670"], ["Togo", ["africa"], "tg", "228"], ["Tonga", ["oceania"], "to", "676"], ["Trinidad and Tobago", ["america", "carribean"], "tt", "1868"], ["Tunisia", ["africa", "north-africa"], "tn", "216"], ["Turkey", ["europe"], "tr", "90", "... ... .. .."], ["Turkmenistan", ["asia", "ex-ussr"], "tm", "993"], ["Tuvalu", ["asia"], "tv", "688"], ["Uganda", ["africa"], "ug", "256"], ["Ukraine", ["europe", "ex-ussr"], "ua", "380", "(..) ... .. .."], ["United Arab Emirates", ["middle-east"], "ae", "971"], ["United Kingdom", ["europe", "eu-union"], "gb", "44", ".... ......"], ["United States", ["america", "north-america"], "us", "1", "(...) ...-....", 0, ["907", "205", "251", "256", "334", "479", "501", "870", "480", "520", "602", "623", "928", "209", "213", "310", "323", "408", "415", "510", "530", "559", "562", "619", "626", "650", "661", "707", "714", "760", "805", "818", "831", "858", "909", "916", "925", "949", "951", "303", "719", "970", "203", "860", "202", "302", "239", "305", "321", "352", "386", "407", "561", "727", "772", "813", "850", "863", "904", "941", "954", "229", "404", "478", "706", "770", "912", "808", "319", "515", "563", "641", "712", "208", "217", "309", "312", "618", "630", "708", "773", "815", "847", "219", "260", "317", "574", "765", "812", "316", "620", "785", "913", "270", "502", "606", "859", "225", "318", "337", "504", "985", "413", "508", "617", "781", "978", "301", "410", "207", "231", "248", "269", "313", "517", "586", "616", "734", "810", "906", "989", "218", "320", "507", "612", "651", "763", "952", "314", "417", "573", "636", "660", "816", "228", "601", "662", "406", "252", "336", "704", "828", "910", "919", "701", "308", "402", "603", "201", "609", "732", "856", "908", "973", "505", "575", "702", "775", "212", "315", "516", "518", "585", "607", "631", "716", "718", "845", "914", "216", "330", "419", "440", "513", "614", "740", "937", "405", "580", "918", "503", "541", "215", "412", "570", "610", "717", "724", "814", "401", "803", "843", "864", "605", "423", "615", "731", "865", "901", "931", "210", "214", "254", "281", "325", "361", "409", "432", "512", "713", "806", "817", "830", "903", "915", "936", "940", "956", "972", "979", "435", "801", "276", "434", "540", "703", "757", "804", "802", "206", "253", "360", "425", "509", "262", "414", "608", "715", "920", "304", "307"]], ["Uruguay", ["america", "south-america"], "uy", "598"], ["Uzbekistan", ["asia", "ex-ussr"], "uz", "998", ".. ... .. .."], ["Vanuatu", ["oceania"], "vu", "678"], ["Vatican City", ["europe"], "va", "39", ".. .... ....", 1], ["Venezuela", ["america", "south-america"], "ve", "58"], ["Vietnam", ["asia"], "vn", "84"], ["Yemen", ["middle-east"], "ye", "967"], ["Zambia", ["africa"], "zm", "260"], ["Zimbabwe", ["africa"], "zw", "263"]], we = [["American Samoa", ["oceania"], "as", "1684"], ["Anguilla", ["america", "carribean"], "ai", "1264"], ["Bermuda", ["america", "north-america"], "bm", "1441"], ["British Virgin Islands", ["america", "carribean"], "vg", "1284"], ["Cayman Islands", ["america", "carribean"], "ky", "1345"], ["Cook Islands", ["oceania"], "ck", "682"], ["Falkland Islands", ["america", "south-america"], "fk", "500"], ["Faroe Islands", ["europe"], "fo", "298"], ["Gibraltar", ["europe"], "gi", "350"], ["Greenland", ["america"], "gl", "299"], ["Jersey", ["europe", "eu-union"], "je", "44", ".... ......"], ["Montserrat", ["america", "carribean"], "ms", "1664"], ["Niue", ["asia"], "nu", "683"], ["Norfolk Island", ["oceania"], "nf", "672"], ["Northern Mariana Islands", ["oceania"], "mp", "1670"], ["Saint Barthélemy", ["america", "carribean"], "bl", "590", "", 1], ["Saint Helena", ["africa"], "sh", "290"], ["Saint Martin", ["america", "carribean"], "mf", "590", "", 2], ["Saint Pierre and Miquelon", ["america", "north-america"], "pm", "508"], ["Sint Maarten", ["america", "carribean"], "sx", "1721"], ["Tokelau", ["oceania"], "tk", "690"], ["Turks and Caicos Islands", ["america", "carribean"], "tc", "1649"], ["U.S. Virgin Islands", ["america", "carribean"], "vi", "1340"], ["Wallis and Futuna", ["oceania"], "wf", "681"]];
  function s(m, c, x, A, f) {
    return !x || f ? m + "".padEnd(c.length, ".") + " " + A : m + "".padEnd(c.length, ".") + " " + x;
  }
  function y(m, c, x, A, f) {
    var t, z, M = [];
    return z = c === !0, [(t = []).concat.apply(t, F(m.map(function(L) {
      var re = { name: L[0], regions: L[1], iso2: L[2], countryCode: L[3], dialCode: L[3], format: s(x, L[3], L[4], A, f), priority: L[5] || 0 }, Y = [];
      return L[6] && L[6].map(function(ne) {
        var ce = function(de) {
          for (var ae = 1; ae < arguments.length; ae++) {
            var n = arguments[ae] != null ? arguments[ae] : {}, o = Object.keys(n);
            typeof Object.getOwnPropertySymbols == "function" && (o = o.concat(Object.getOwnPropertySymbols(n).filter(function(d) {
              return Object.getOwnPropertyDescriptor(n, d).enumerable;
            }))), o.forEach(function(d) {
              u(de, d, n[d]);
            });
          }
          return de;
        }({}, re);
        ce.dialCode = L[3] + ne, ce.isAreaCode = !0, ce.areaCodeLength = ne.length, Y.push(ce);
      }), Y.length > 0 ? (re.mainCode = !0, z || c.constructor.name === "Array" && c.includes(L[2]) ? (re.hasAreaCodes = !0, [re].concat(Y)) : (M = M.concat(Y), [re])) : [re];
    }))), M];
  }
  function C(m, c, x, A) {
    if (x !== null) {
      var f = Object.keys(x), t = Object.values(x);
      f.forEach(function(z, M) {
        if (A)
          return m.push([z, t[M]]);
        var L = m.findIndex(function(Y) {
          return Y[0] === z;
        });
        if (L === -1) {
          var re = [z];
          re[c] = t[M], m.push(re);
        } else
          m[L][c] = t[M];
      });
    }
  }
  function E(m, c) {
    return c.length === 0 ? m : m.map(function(x) {
      var A = c.findIndex(function(t) {
        return t[0] === x[2];
      });
      if (A === -1)
        return x;
      var f = c[A];
      return f[1] && (x[4] = f[1]), f[3] && (x[5] = f[3]), f[2] && (x[6] = f[2]), x;
    });
  }
  var fe = function m(c, x, A, f, t, z, M, L, re, Y, ne, ce, de, ae) {
    te(this, m), this.filterRegions = function(p, S) {
      if (typeof p == "string") {
        var P = p;
        return S.filter(function(h) {
          return h.regions.some(function(v) {
            return v === P;
          });
        });
      }
      return S.filter(function(h) {
        return p.map(function(v) {
          return h.regions.some(function(X) {
            return X === v;
          });
        }).some(function(v) {
          return v;
        });
      });
    }, this.sortTerritories = function(p, S) {
      var P = [].concat(F(p), F(S));
      return P.sort(function(h, v) {
        return h.name < v.name ? -1 : h.name > v.name ? 1 : 0;
      }), P;
    }, this.getFilteredCountryList = function(p, S, P) {
      return p.length === 0 ? S : P ? p.map(function(h) {
        var v = S.find(function(X) {
          return X.iso2 === h;
        });
        if (v)
          return v;
      }).filter(function(h) {
        return h;
      }) : S.filter(function(h) {
        return p.some(function(v) {
          return v === h.iso2;
        });
      });
    }, this.localizeCountries = function(p, S, P) {
      for (var h = 0; h < p.length; h++)
        S[p[h].iso2] !== void 0 ? p[h].localName = S[p[h].iso2] : S[p[h].name] !== void 0 && (p[h].localName = S[p[h].name]);
      return P || p.sort(function(v, X) {
        return v.localName < X.localName ? -1 : v.localName > X.localName ? 1 : 0;
      }), p;
    }, this.getCustomAreas = function(p, S) {
      for (var P = [], h = 0; h < S.length; h++) {
        var v = JSON.parse(JSON.stringify(p));
        v.dialCode += S[h], P.push(v);
      }
      return P;
    }, this.excludeCountries = function(p, S) {
      return S.length === 0 ? p : p.filter(function(P) {
        return !S.includes(P.iso2);
      });
    };
    var n = function(p, S, P) {
      var h = [];
      return C(h, 1, p, !0), C(h, 3, S), C(h, 2, P), h;
    }(L, re, Y), o = E(JSON.parse(JSON.stringify(me)), n), d = E(JSON.parse(JSON.stringify(we)), n), k = Ce(y(o, c, ce, de, ae), 2), N = k[0], T = k[1];
    if (x) {
      var D = Ce(y(d, c, ce, de, ae), 2), U = D[0];
      D[1], N = this.sortTerritories(U, N);
    }
    A && (N = this.filterRegions(A, N)), this.onlyCountries = this.localizeCountries(this.excludeCountries(this.getFilteredCountryList(f, N, M.includes("onlyCountries")), z), ne, M.includes("onlyCountries")), this.preferredCountries = t.length === 0 ? [] : this.localizeCountries(this.getFilteredCountryList(t, N, M.includes("preferredCountries")), ne, M.includes("preferredCountries")), this.hiddenAreaCodes = this.excludeCountries(this.getFilteredCountryList(f, T), z);
  }, Ae = function(m) {
    function c(f) {
      var t;
      te(this, c), (t = function(n, o) {
        return !o || ge(o) !== "object" && typeof o != "function" ? ve(n) : o;
      }(this, Oe(c).call(this, f))).getProbableCandidate = j()(function(n) {
        return n && n.length !== 0 ? t.state.onlyCountries.filter(function(o) {
          return G()(o.name.toLowerCase(), n.toLowerCase());
        }, ve(ve(t)))[0] : null;
      }), t.guessSelectedCountry = j()(function(n, o, d, k) {
        var N;
        if (t.props.enableAreaCodes === !1 && (k.some(function(U) {
          if (G()(n, U.dialCode))
            return d.some(function(p) {
              if (U.iso2 === p.iso2 && p.mainCode)
                return N = p, !0;
            }), !0;
        }), N))
          return N;
        var T = d.find(function(U) {
          return U.iso2 == o;
        });
        if (n.trim() === "")
          return T;
        var D = d.reduce(function(U, p) {
          return G()(n, p.dialCode) && (p.dialCode.length > U.dialCode.length || p.dialCode.length === U.dialCode.length && p.priority < U.priority) ? p : U;
        }, { dialCode: "", priority: 10001 }, ve(ve(t)));
        return D.name ? D : T;
      }), t.updateCountry = function(n) {
        var o, d = t.state.onlyCountries;
        (o = n.indexOf(0) >= "0" && n.indexOf(0) <= "9" ? d.find(function(k) {
          return k.dialCode == +n;
        }) : d.find(function(k) {
          return k.iso2 == n;
        })) && o.dialCode && t.setState({ selectedCountry: o, formattedNumber: t.props.disableCountryCode ? "" : t.formatNumber(o.dialCode, o) });
      }, t.scrollTo = function(n, o) {
        if (n) {
          var d = t.dropdownRef;
          if (d && document.body) {
            var k = d.offsetHeight, N = d.getBoundingClientRect().top + document.body.scrollTop, T = N + k, D = n, U = D.getBoundingClientRect(), p = D.offsetHeight, S = U.top + document.body.scrollTop, P = S + p, h = S - N + d.scrollTop, v = k / 2 - p / 2;
            if (t.props.enableSearch ? S < N + 32 : S < N)
              o && (h -= v), d.scrollTop = h;
            else if (P > T) {
              o && (h += v);
              var X = k - p;
              d.scrollTop = h - X;
            }
          }
        }
      }, t.scrollToTop = function() {
        var n = t.dropdownRef;
        n && document.body && (n.scrollTop = 0);
      }, t.formatNumber = function(n, o) {
        if (!o)
          return n;
        var d, k = o.format, N = t.props, T = N.disableCountryCode, D = N.enableAreaCodeStretch, U = N.enableLongNumbers, p = N.autoFormat;
        if (T ? ((d = k.split(" ")).shift(), d = d.join(" ")) : D && o.isAreaCode ? ((d = k.split(" "))[1] = d[1].replace(/\.+/, "".padEnd(o.areaCodeLength, ".")), d = d.join(" ")) : d = k, !n || n.length === 0)
          return T ? "" : t.props.prefix;
        if (n && n.length < 2 || !d || !p)
          return T ? n : t.props.prefix + n;
        var S, P = q()(d, function(h, v) {
          if (h.remainingText.length === 0)
            return h;
          if (v !== ".")
            return { formattedText: h.formattedText + v, remainingText: h.remainingText };
          var X, je = ie(X = h.remainingText) || O(X) || se(), ye = je[0], Te = je.slice(1);
          return { formattedText: h.formattedText + ye, remainingText: Te };
        }, { formattedText: "", remainingText: n.split("") });
        return (S = U ? P.formattedText + P.remainingText.join("") : P.formattedText).includes("(") && !S.includes(")") && (S += ")"), S;
      }, t.cursorToEnd = function() {
        var n = t.numberInputRef;
        if (document.activeElement === n) {
          n.focus();
          var o = n.value.length;
          n.value.charAt(o - 1) === ")" && (o -= 1), n.setSelectionRange(o, o);
        }
      }, t.getElement = function(n) {
        return t["flag_no_".concat(n)];
      }, t.getCountryData = function() {
        return t.state.selectedCountry ? { name: t.state.selectedCountry.name || "", dialCode: t.state.selectedCountry.dialCode || "", countryCode: t.state.selectedCountry.iso2 || "", format: t.state.selectedCountry.format || "" } : {};
      }, t.handleFlagDropdownClick = function(n) {
        if (n.preventDefault(), t.state.showDropdown || !t.props.disabled) {
          var o = t.state, d = o.preferredCountries, k = o.onlyCountries, N = o.selectedCountry, T = t.concatPreferredCountries(d, k).findIndex(function(D) {
            return D.dialCode === N.dialCode && D.iso2 === N.iso2;
          });
          t.setState({ showDropdown: !t.state.showDropdown, highlightCountryIndex: T }, function() {
            t.state.showDropdown && t.scrollTo(t.getElement(t.state.highlightCountryIndex));
          });
        }
      }, t.handleInput = function(n) {
        var o = n.target.value, d = t.props, k = d.prefix, N = d.onChange, T = t.props.disableCountryCode ? "" : k, D = t.state.selectedCountry, U = t.state.freezeSelection;
        if (!t.props.countryCodeEditable) {
          var p = k + (D.hasAreaCodes ? t.state.onlyCountries.find(function(pe) {
            return pe.iso2 === D.iso2 && pe.mainCode;
          }).dialCode : D.dialCode);
          if (o.slice(0, p.length) !== p)
            return;
        }
        if (o === k)
          return N && N("", t.getCountryData(), n, ""), t.setState({ formattedNumber: "" });
        if (!(o.replace(/\D/g, "").length > 15 && (t.props.enableLongNumbers === !1 || typeof t.props.enableLongNumbers == "number" && o.replace(/\D/g, "").length > t.props.enableLongNumbers)) && o !== t.state.formattedNumber) {
          n.preventDefault ? n.preventDefault() : n.returnValue = !1;
          var S = t.props.country, P = t.state, h = P.onlyCountries, v = P.selectedCountry, X = P.hiddenAreaCodes;
          if (N && n.persist(), o.length > 0) {
            var je = o.replace(/\D/g, "");
            (!t.state.freezeSelection || v && v.dialCode.length > je.length) && (D = t.props.disableCountryGuess ? v : t.guessSelectedCountry(je.substring(0, 6), S, h, X) || v, U = !1), T = t.formatNumber(je, D), D = D.dialCode ? D : v;
          }
          var ye = n.target.selectionStart, Te = n.target.selectionStart, Pe = t.state.formattedNumber, Ge = T.length - Pe.length;
          t.setState({ formattedNumber: T, freezeSelection: U, selectedCountry: D }, function() {
            Ge > 0 && (Te -= Ge), T.charAt(T.length - 1) == ")" ? t.numberInputRef.setSelectionRange(T.length - 1, T.length - 1) : Te > 0 && Pe.length >= T.length ? t.numberInputRef.setSelectionRange(Te, Te) : ye < Pe.length && t.numberInputRef.setSelectionRange(ye, ye), N && N(T.replace(/[^0-9]+/g, ""), t.getCountryData(), n, T);
          });
        }
      }, t.handleInputClick = function(n) {
        t.setState({ showDropdown: !1 }), t.props.onClick && t.props.onClick(n, t.getCountryData());
      }, t.handleDoubleClick = function(n) {
        var o = n.target.value.length;
        n.target.setSelectionRange(0, o);
      }, t.handleFlagItemClick = function(n, o) {
        var d = t.state.selectedCountry, k = t.state.onlyCountries.find(function(U) {
          return U == n;
        });
        if (k) {
          var N = t.state.formattedNumber.replace(" ", "").replace("(", "").replace(")", "").replace("-", ""), T = N.length > 1 ? N.replace(d.dialCode, k.dialCode) : k.dialCode, D = t.formatNumber(T.replace(/\D/g, ""), k);
          t.setState({ showDropdown: !1, selectedCountry: k, freezeSelection: !0, formattedNumber: D, searchValue: "" }, function() {
            t.cursorToEnd(), t.props.onChange && t.props.onChange(D.replace(/[^0-9]+/g, ""), t.getCountryData(), o, D);
          });
        }
      }, t.handleInputFocus = function(n) {
        t.numberInputRef && t.numberInputRef.value === t.props.prefix && t.state.selectedCountry && !t.props.disableCountryCode && t.setState({ formattedNumber: t.props.prefix + t.state.selectedCountry.dialCode }, function() {
          t.props.jumpCursorToEnd && setTimeout(t.cursorToEnd, 0);
        }), t.setState({ placeholder: "" }), t.props.onFocus && t.props.onFocus(n, t.getCountryData()), t.props.jumpCursorToEnd && setTimeout(t.cursorToEnd, 0);
      }, t.handleInputBlur = function(n) {
        n.target.value || t.setState({ placeholder: t.props.placeholder }), t.props.onBlur && t.props.onBlur(n, t.getCountryData());
      }, t.handleInputCopy = function(n) {
        if (t.props.copyNumbersOnly) {
          var o = window.getSelection().toString().replace(/[^0-9]+/g, "");
          n.clipboardData.setData("text/plain", o), n.preventDefault();
        }
      }, t.getHighlightCountryIndex = function(n) {
        var o = t.state.highlightCountryIndex + n;
        return o < 0 || o >= t.state.onlyCountries.length + t.state.preferredCountries.length ? o - n : t.props.enableSearch && o > t.getSearchFilteredCountries().length ? 0 : o;
      }, t.searchCountry = function() {
        var n = t.getProbableCandidate(t.state.queryString) || t.state.onlyCountries[0], o = t.state.onlyCountries.findIndex(function(d) {
          return d == n;
        }) + t.state.preferredCountries.length;
        t.scrollTo(t.getElement(o), !0), t.setState({ queryString: "", highlightCountryIndex: o });
      }, t.handleKeydown = function(n) {
        var o = t.props.keys, d = n.target.className;
        if (d.includes("selected-flag") && n.which === o.ENTER && !t.state.showDropdown)
          return t.handleFlagDropdownClick(n);
        if (d.includes("form-control") && (n.which === o.ENTER || n.which === o.ESC))
          return n.target.blur();
        if (t.state.showDropdown && !t.props.disabled && (!d.includes("search-box") || n.which === o.UP || n.which === o.DOWN || n.which === o.ENTER || n.which === o.ESC && n.target.value === "")) {
          n.preventDefault ? n.preventDefault() : n.returnValue = !1;
          var k = function(N) {
            t.setState({ highlightCountryIndex: t.getHighlightCountryIndex(N) }, function() {
              t.scrollTo(t.getElement(t.state.highlightCountryIndex), !0);
            });
          };
          switch (n.which) {
            case o.DOWN:
              k(1);
              break;
            case o.UP:
              k(-1);
              break;
            case o.ENTER:
              t.props.enableSearch ? t.handleFlagItemClick(t.getSearchFilteredCountries()[t.state.highlightCountryIndex] || t.getSearchFilteredCountries()[0], n) : t.handleFlagItemClick([].concat(F(t.state.preferredCountries), F(t.state.onlyCountries))[t.state.highlightCountryIndex], n);
              break;
            case o.ESC:
            case o.TAB:
              t.setState({ showDropdown: !1 }, t.cursorToEnd);
              break;
            default:
              (n.which >= o.A && n.which <= o.Z || n.which === o.SPACE) && t.setState({ queryString: t.state.queryString + String.fromCharCode(n.which) }, t.state.debouncedQueryStingSearcher);
          }
        }
      }, t.handleInputKeyDown = function(n) {
        var o = t.props, d = o.keys, k = o.onEnterKeyPress, N = o.onKeyDown;
        n.which === d.ENTER && k && k(n), N && N(n);
      }, t.handleClickOutside = function(n) {
        t.dropdownRef && !t.dropdownContainerRef.contains(n.target) && t.state.showDropdown && t.setState({ showDropdown: !1 });
      }, t.handleSearchChange = function(n) {
        var o = n.currentTarget.value, d = t.state, k = d.preferredCountries, N = d.selectedCountry, T = 0;
        if (o === "" && N) {
          var D = t.state.onlyCountries;
          T = t.concatPreferredCountries(k, D).findIndex(function(U) {
            return U == N;
          }), setTimeout(function() {
            return t.scrollTo(t.getElement(T));
          }, 100);
        }
        t.setState({ searchValue: o, highlightCountryIndex: T });
      }, t.concatPreferredCountries = function(n, o) {
        return n.length > 0 ? F(new Set(n.concat(o))) : o;
      }, t.getDropdownCountryName = function(n) {
        return n.localName || n.name;
      }, t.getSearchFilteredCountries = function() {
        var n = t.state, o = n.preferredCountries, d = n.onlyCountries, k = n.searchValue, N = t.props.enableSearch, T = t.concatPreferredCountries(o, d), D = k.trim().toLowerCase().replace("+", "");
        if (N && D) {
          if (/^\d+$/.test(D))
            return T.filter(function(S) {
              var P = S.dialCode;
              return ["".concat(P)].some(function(h) {
                return h.toLowerCase().includes(D);
              });
            });
          var U = T.filter(function(S) {
            var P = S.iso2;
            return ["".concat(P)].some(function(h) {
              return h.toLowerCase().includes(D);
            });
          }), p = T.filter(function(S) {
            var P = S.name, h = S.localName;
            return S.iso2, ["".concat(P), "".concat(h || "")].some(function(v) {
              return v.toLowerCase().includes(D);
            });
          });
          return t.scrollToTop(), F(new Set([].concat(U, p)));
        }
        return T;
      }, t.getCountryDropdownList = function() {
        var n = t.state, o = n.preferredCountries, d = n.highlightCountryIndex, k = n.showDropdown, N = n.searchValue, T = t.props, D = T.disableDropdown, U = T.prefix, p = t.props, S = p.enableSearch, P = p.searchNotFound, h = p.disableSearchIcon, v = p.searchClass, X = p.searchStyle, je = p.searchPlaceholder, ye = p.autocompleteSearch, Te = t.getSearchFilteredCountries().map(function(pe, Ue) {
          var We = d === Ue, dt = K()({ country: !0, preferred: pe.iso2 === "us" || pe.iso2 === "gb", active: pe.iso2 === "us", highlight: We }), Je = "flag ".concat(pe.iso2);
          return g.a.createElement("li", Object.assign({ ref: function(Ze) {
            return t["flag_no_".concat(Ue)] = Ze;
          }, key: "flag_no_".concat(Ue), "data-flag-key": "flag_no_".concat(Ue), className: dt, "data-dial-code": "1", tabIndex: D ? "-1" : "0", "data-country-code": pe.iso2, onClick: function(Ze) {
            return t.handleFlagItemClick(pe, Ze);
          }, role: "option" }, We ? { "aria-selected": !0 } : {}), g.a.createElement("div", { className: Je }), g.a.createElement("span", { className: "country-name" }, t.getDropdownCountryName(pe)), g.a.createElement("span", { className: "dial-code" }, pe.format ? t.formatNumber(pe.dialCode, pe) : U + pe.dialCode));
        }), Pe = g.a.createElement("li", { key: "dashes", className: "divider" });
        o.length > 0 && (!S || S && !N.trim()) && Te.splice(o.length, 0, Pe);
        var Ge = K()(u({ "country-list": !0, hide: !k }, t.props.dropdownClass, !0));
        return g.a.createElement("ul", { ref: function(pe) {
          return !S && pe && pe.focus(), t.dropdownRef = pe;
        }, className: Ge, style: t.props.dropdownStyle, role: "listbox", tabIndex: "0" }, S && g.a.createElement("li", { className: K()(u({ search: !0 }, v, v)) }, !h && g.a.createElement("span", { className: K()(u({ "search-emoji": !0 }, "".concat(v, "-emoji"), v)), role: "img", "aria-label": "Magnifying glass" }, "🔎"), g.a.createElement("input", { className: K()(u({ "search-box": !0 }, "".concat(v, "-box"), v)), style: X, type: "search", placeholder: je, autoFocus: !0, autoComplete: ye ? "on" : "off", value: N, onChange: t.handleSearchChange })), Te.length > 0 ? Te : g.a.createElement("li", { className: "no-entries-message" }, g.a.createElement("span", null, P)));
      };
      var z, M = new fe(f.enableAreaCodes, f.enableTerritories, f.regions, f.onlyCountries, f.preferredCountries, f.excludeCountries, f.preserveOrder, f.masks, f.priority, f.areaCodes, f.localization, f.prefix, f.defaultMask, f.alwaysDefaultMask), L = M.onlyCountries, re = M.preferredCountries, Y = M.hiddenAreaCodes, ne = f.value ? f.value.replace(/\D/g, "") : "";
      z = f.disableInitialCountryGuess ? 0 : ne.length > 1 ? t.guessSelectedCountry(ne.substring(0, 6), f.country, L, Y) || 0 : f.country && L.find(function(n) {
        return n.iso2 == f.country;
      }) || 0;
      var ce, de = ne.length < 2 && z && !G()(ne, z.dialCode) ? z.dialCode : "";
      ce = ne === "" && z === 0 ? "" : t.formatNumber((f.disableCountryCode ? "" : de) + ne, z.name ? z : void 0);
      var ae = L.findIndex(function(n) {
        return n == z;
      });
      return t.state = { showDropdown: f.showDropdown, formattedNumber: ce, onlyCountries: L, preferredCountries: re, hiddenAreaCodes: Y, selectedCountry: z, highlightCountryIndex: ae, queryString: "", freezeSelection: !1, debouncedQueryStingSearcher: R()(t.searchCountry, 250), searchValue: "" }, t;
    }
    var x, A;
    return function(f, t) {
      if (typeof t != "function" && t !== null)
        throw new TypeError("Super expression must either be null or a function");
      f.prototype = Object.create(t && t.prototype, { constructor: { value: f, writable: !0, configurable: !0 } }), t && Ee(f, t);
    }(c, m), x = c, (A = [{ key: "componentDidMount", value: function() {
      document.addEventListener && this.props.enableClickOutside && document.addEventListener("mousedown", this.handleClickOutside), this.props.onMount && this.props.onMount(this.state.formattedNumber.replace(/[^0-9]+/g, ""), this.getCountryData(), this.state.formattedNumber);
    } }, { key: "componentWillUnmount", value: function() {
      document.removeEventListener && this.props.enableClickOutside && document.removeEventListener("mousedown", this.handleClickOutside);
    } }, { key: "componentDidUpdate", value: function(f, t, z) {
      f.country !== this.props.country ? this.updateCountry(this.props.country) : f.value !== this.props.value && this.updateFormattedNumber(this.props.value);
    } }, { key: "updateFormattedNumber", value: function(f) {
      if (f === null)
        return this.setState({ selectedCountry: 0, formattedNumber: "" });
      var t = this.state, z = t.onlyCountries, M = t.selectedCountry, L = t.hiddenAreaCodes, re = this.props, Y = re.country, ne = re.prefix;
      if (f === "")
        return this.setState({ selectedCountry: M, formattedNumber: "" });
      var ce, de, ae = f.replace(/\D/g, "");
      if (M && G()(f, ne + M.dialCode))
        de = this.formatNumber(ae, M), this.setState({ formattedNumber: de });
      else {
        var n = (ce = this.props.disableCountryGuess ? M : this.guessSelectedCountry(ae.substring(0, 6), Y, z, L) || M) && G()(ae, ne + ce.dialCode) ? ce.dialCode : "";
        de = this.formatNumber((this.props.disableCountryCode ? "" : n) + ae, ce || void 0), this.setState({ selectedCountry: ce, formattedNumber: de });
      }
    } }, { key: "render", value: function() {
      var f, t, z, M = this, L = this.state, re = L.onlyCountries, Y = L.selectedCountry, ne = L.showDropdown, ce = L.formattedNumber, de = L.hiddenAreaCodes, ae = this.props, n = ae.disableDropdown, o = ae.renderStringAsFlag, d = ae.isValid, k = ae.defaultErrorMessage, N = ae.specialLabel;
      if (typeof d == "boolean")
        t = d;
      else {
        var T = d(ce.replace(/\D/g, ""), Y, re, de);
        typeof T == "boolean" ? (t = T) === !1 && (z = k) : (t = !1, z = T);
      }
      var D = K()((u(f = {}, this.props.containerClass, !0), u(f, "react-tel-input", !0), f)), U = K()({ arrow: !0, up: ne }), p = K()(u({ "form-control": !0, "invalid-number": !t, open: ne }, this.props.inputClass, !0)), S = K()({ "selected-flag": !0, open: ne }), P = K()(u({ "flag-dropdown": !0, "invalid-number": !t, open: ne }, this.props.buttonClass, !0)), h = "flag ".concat(Y && Y.iso2);
      return g.a.createElement("div", { className: "".concat(D, " ").concat(this.props.className), style: this.props.style || this.props.containerStyle, onKeyDown: this.handleKeydown }, N && g.a.createElement("div", { className: "special-label" }, N), z && g.a.createElement("div", { className: "invalid-number-message" }, z), g.a.createElement("input", Object.assign({ className: p, style: this.props.inputStyle, onChange: this.handleInput, onClick: this.handleInputClick, onDoubleClick: this.handleDoubleClick, onFocus: this.handleInputFocus, onBlur: this.handleInputBlur, onCopy: this.handleInputCopy, value: ce, onKeyDown: this.handleInputKeyDown, placeholder: this.props.placeholder, disabled: this.props.disabled, type: "tel" }, this.props.inputProps, { ref: function(v) {
        M.numberInputRef = v, typeof M.props.inputProps.ref == "function" ? M.props.inputProps.ref(v) : typeof M.props.inputProps.ref == "object" && (M.props.inputProps.ref.current = v);
      } })), g.a.createElement("div", { className: P, style: this.props.buttonStyle, ref: function(v) {
        return M.dropdownContainerRef = v;
      } }, o ? g.a.createElement("div", { className: S }, o) : g.a.createElement("div", { onClick: n ? void 0 : this.handleFlagDropdownClick, className: S, title: Y ? "".concat(Y.localName || Y.name, ": + ").concat(Y.dialCode) : "", tabIndex: n ? "-1" : "0", role: "button", "aria-haspopup": "listbox", "aria-expanded": !!ne || void 0 }, g.a.createElement("div", { className: h }, !n && g.a.createElement("div", { className: U }))), ne && this.getCountryDropdownList()));
    } }]) && Se(x.prototype, A), c;
  }(g.a.Component);
  Ae.defaultProps = { country: "", value: "", onlyCountries: [], preferredCountries: [], excludeCountries: [], placeholder: "1 (702) 123-4567", searchPlaceholder: "search", searchNotFound: "No entries to show", flagsImagePath: "./flags.png", disabled: !1, containerStyle: {}, inputStyle: {}, buttonStyle: {}, dropdownStyle: {}, searchStyle: {}, containerClass: "", inputClass: "", buttonClass: "", dropdownClass: "", searchClass: "", className: "", autoFormat: !0, enableAreaCodes: !1, enableTerritories: !1, disableCountryCode: !1, disableDropdown: !1, enableLongNumbers: !1, countryCodeEditable: !0, enableSearch: !1, disableSearchIcon: !1, disableInitialCountryGuess: !1, disableCountryGuess: !1, regions: "", inputProps: {}, localization: {}, masks: null, priority: null, areaCodes: null, preserveOrder: [], defaultMask: "... ... ... ... ..", alwaysDefaultMask: !1, prefix: "+", copyNumbersOnly: !0, renderStringAsFlag: "", autocompleteSearch: !1, jumpCursorToEnd: !0, enableAreaCodeStretch: !1, enableClickOutside: !0, showDropdown: !1, isValid: !0, defaultErrorMessage: "", specialLabel: "Phone", onEnterKeyPress: null, keys: { UP: 38, DOWN: 40, RIGHT: 39, LEFT: 37, ENTER: 13, ESC: 27, PLUS: 43, A: 65, Z: 90, SPACE: 32, TAB: 9 } }, Q.default = Ae;
}]);
const qt = /* @__PURE__ */ Bt(Ut);
function Yt({
  name: Z,
  className: Q = "",
  onChange: l,
  value: u,
  ...O
}) {
  const F = Ht();
  return /* @__PURE__ */ Vt(qt, {
    ...O,
    value: u ? u.trim().replace(/^\+?/, "+") : void 0,
    country: u ? void 0 : F,
    enableSearch: !0,
    disableSearchIcon: !0,
    inputProps: {
      name: Z,
      required: O.required,
      placeholder: O.placeholder
    },
    onChange: (ie) => {
      l(`+${ie}`);
    },
    containerClass: Gt("hover:border-emphasis dark:focus:border-emphasis border-default !bg-default rounded-md border focus-within:outline-none focus-within:ring-2 focus-within:ring-brand-default disabled:cursor-not-allowed", Q),
    inputClass: "text-sm focus:ring-0 !bg-default text-default",
    buttonClass: "text-emphasis !bg-default hover:!bg-emphasis",
    searchClass: "!text-default !bg-default hover:!bg-emphasis",
    dropdownClass: "!text-default !bg-default",
    inputStyle: {
      width: "inherit",
      border: 0
    },
    searchStyle: {
      display: "flex",
      flexDirection: "row",
      alignItems: "center",
      padding: "6px 12px",
      gap: "8px",
      width: "296px",
      height: "28px",
      marginLeft: "-4px"
    },
    dropdownStyle: {
      width: "max-content"
    }
  });
}
const Ht = () => {
  const [Z, Q] = Rt("us"), l = $t.viewer.public.countryCode.useQuery(void 0, {
    refetchOnWindowFocus: !1,
    refetchOnReconnect: !1,
    retry: !1
  });
  return zt(function() {
    var F;
    const O = l.data;
    O != null && O.countryCode && (Kt(O == null ? void 0 : O.countryCode) ? Q(O.countryCode.toLowerCase()) : Q(((F = navigator.language.split("-")[1]) == null ? void 0 : F.toLowerCase()) || "us"));
  }, [l.data]), Z;
};
export {
  Yt as default
};
