// @calcom/atoms/dist/PoweredBy-LfOhjGas.js
import { an as c, ao as m, ap as d, aq as p, ar as x, as as s } from "./index-6PQ0n_zn.js";
import { jsx as a, jsxs as t, Fragment as o } from "react/jsx-runtime";
const b = ({
  logoOnly: i
}) => {
  const {
    t: l
  } = c(), e = m.useSession(), n = d(), r = e.data ? e.data.hasValidLicense : null;
  return /* @__PURE__ */ a("div", {
    className: `p-2 text-center text-xs sm:text-right${n ? " max-w-3xl" : ""}`,
    children: /* @__PURE__ */ t(p, {
      href: x,
      target: "_blank",
      className: "text-subtle",
      children: [!i && /* @__PURE__ */ t(o, {
        children: [l("powered_by"), " "]
      }), s === "Cal.com" || !r ? /* @__PURE__ */ a(o, {
        children: /* @__PURE__ */ a("img", {
          className: "relative -mt-px inline h-[10px] w-auto dark:invert",
          src: "/api/logo",
          alt: "Cal.com Logo"
        })
      }) : /* @__PURE__ */ a("span", {
        className: "text-emphasis font-semibold opacity-50 hover:opacity-100",
        children: s
      })]
    })
  });
};
export {
  b as default
};
